backend/src/test/java/com/nikogrid/backend/repositories/UserRepositoryTest.java

```diff
@@ -0,0 +1,251 @@
+        assertThat(savedUser.getRoles()).containsExactly(adminRole);
+        assertThat(savedUser.getCreated_at()).isNotNull();
+    }
+
+    @Test
+    void testFindByEmail_ExistingUser() {
+        userRepository.save(testUser1);
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findByEmail("test1@example.com");
+
+        assertThat(foundUser).isPresent();
+        assertThat(foundUser.get().getEmail()).isEqualTo("test1@example.com");
+        assertThat(foundUser.get().getPassword()).isEqualTo("password123");
+    }
+
+    @Test
+    void testFindByEmail_NonExistingUser() {
+        Optional<User> foundUser = userRepository.findByEmail("nonexistent@example.com");
+
+        assertThat(foundUser).isNotPresent();
+    }
+
+    @Test
+    void testFindByEmail_CaseSensitive() {
+        userRepository.save(testUser1);
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findByEmail("TEST1@EXAMPLE.COM");
+
+        assertThat(foundUser).isNotPresent();
+    }
+
+    @Test
+    void testFindById() {
+        User savedUser = userRepository.save(testUser1);
+        UUID userId = savedUser.getId();
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findById(userId);
+
+        assertThat(foundUser).isPresent();
+        assertThat(foundUser.get().getId()).isEqualTo(userId);
+        assertThat(foundUser.get().getEmail()).isEqualTo("test1@example.com");
+    }
+
+    @Test
+    void testFindAll() {
+        userRepository.save(testUser1);
+        userRepository.save(testUser2);
+        entityManager.flush();
+
+        List<User> allUsers = userRepository.findAll();
+
+        assertThat(allUsers).hasSize(2);
+        assertThat(allUsers).extracting(User::getEmail)
+                .containsExactlyInAnyOrder("test1@example.com", "test2@example.com");
+    }
+
+    @Test
+    void testDeleteUser() {
+        User savedUser = userRepository.save(testUser1);
+        UUID userId = savedUser.getId();
+        entityManager.flush();
+
+        userRepository.deleteById(userId);
+        entityManager.flush();
+
+        Optional<User> foundUser = userRepository.findById(userId);
+        assertThat(foundUser).isNotPresent();
+    }
+
+    @Test
+    void testUpdateUser() {
+        User savedUser = userRepository.save(testUser1);
+        entityManager.flush();
+
+        savedUser.setEmail("updated@example.com");
+        savedUser.setPassword("newPassword");
+
+        User updatedUser = userRepository.save(savedUser);
+        entityManager.flush();
+
+        assertThat(updatedUser.getEmail()).isEqualTo("updated@example.com");
+        assertThat(updatedUser.getPassword()).isEqualTo("newPassword");
+
+        Optional<User> foundUser = userRepository.findByEmail("updated@example.com");
+        assertThat(foundUser).isPresent();
+        assertThat(foundUser.get().getPassword()).isEqualTo("newPassword");
+    }
```

Comment:
These tests provide no benefit as they are testing spring provided functions, which spring should already test itself.